### PR TITLE
[hotfix][table] Convert FlinkCalciteSqlValidator to Java

### PR DESCRIPTION
**Pull request by twalthr**
## What is the purpose of the change

Converts the `SqlValidator` extension to Java. This class is likely to extend Calcite. So it makes sense to have it in Java to quickly copy paste code.

## Brief change log

Class converted to Java.

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+	@Override
+	protected RelDataType getLogicalSourceRowType(RelDataType sourceRowType, SqlInsert insert) {
+		return unwrapTypeFactory(typeFactory).toSql(sourceRowType);
+	}
+
+	@Override
+	protected RelDataType getLogicalTargetRowType(RelDataType targetRowType, SqlInsert insert) {
+		return unwrapTypeFactory(typeFactory).toSql(targetRowType);
+	}
```
Comment (dawidwys) [inline flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java]: Just out of curiosity. Do we need those methods?

Comment (twalthr) [reply flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java]: Good point. I should be needed. I removed it. Will merge it once Travis gives green light.

**Comment by twalthr** [comment]
@flinkbot run travis

**Comment by twalthr** [comment]
Master was broken at that time need to rerun Travis.

@flinkbot run travis

**Comment by twalthr** [comment]
@flinkbot run travis

**Pull request closed without merging**